os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -402,12 +393,12 @@ static int imxrt_erase(FAR struct mtd_dev_s *dev, off_t startsector, size_t nsec
 
 		/* Not using sub-sector erase.  Erase each full sector */
 
-		imxrt_sectorerase(priv, startsector + sectors_erased);
+		imxrt_sectorerase(priv, startblock + sectors_erased);
 		sectors_erased++;
 	}
-	#endif
-	
-	return (int)sectors_erased;
+	fvdbg("erased sectors = %d\n", sectors_erased);
+
+	return sectors_erased == nsectors ? OK : ERROR;
```

Comment:
Number of erased block should be return, -EIO is ok for error case

Reply:
Modified as suggested, thank you!